### PR TITLE
Add syntax group for colors and numbers

### PR DESCRIPTION
**Pull request by Fymyte**
Highlight Numbers and colors as Numbers
Giving colors a group now allows for using https://github.com/ap/vim-css-color types plugins

I am definitely not an expert in pattern matching so I've felt my way along and stopped when it worked
Any advice on how to improve those patterns is very welcome.

Signed-off-by: Fymyte <pierguill@gmail.com>

### CODE REVIEW

**Comment by fladson** [comment]
Hi @Fymyte, thank you for the contribution and sorry for not seeing this earlier.

I will take a look and get back to you.

**Comment by Fymyte** [comment]
I think that for the colors, we should only match something like `#[0-9A-Fa-f]\{6,8}` but I am not sure if it would match colors

**Comment by Fymyte** [comment]
I think I finally managed to restrain color matching. I found what I was doing wrong if you can test this

**Comment by fladson** [comment]
Hey @Fymyte I did some testing today and I am not sure if we are there yet.
Some numbers found in the middle of values are being highlighted but not the entire value, so I am not sure if that is the desired outcome here.
I believe you want to highlight only the values that are numbers, right? If so, we also need to include the possible prefixes (`+, -, etc.`) and suffixes (`%, px, etc.`).

Screenshot 2021-12-21 at 11 36 01

For the theme configuration, it is working just fine:
Screenshot 2021-12-21 at 11 39 23

**Comment by Fymyte** [comment]
Fixed matching for colors with only 3 char, uses vim's builtin match for Hexa characters
Added match for prefix/suffix of number.
Fixed number highlighted in words

In the latter, I had to tweak the `kittyActionKW` match as it matched every whitespaces which followed, thus making impossible for number to use leader whitespace for matching.
It should not break anything from what I've seen

File: syntax/kitty.vim [hunk 1]
```diff
@@ -1,6 +1,10 @@
+syn match kittySt '.*$' contains=kittyNumber,kittyColor
+syn match kittyColor '#\x\{3,8}' contained
+syn match kittyNumber '\s[+-]\?\d\+\.\?\d*\(%\|px\|pt\|em\)\?'ms=s+1 contained contains=kittyUnit
+syn match kittyUnit '\(px\|pt\|em\)' contained
 " Match keywords only at the start of the line. Must come before other rules
 " matching start of line
```
Comment (fladson) [inline syntax/kitty.vim]: Can you also please remove these, since it's not valid anymore.

**Comment by fladson** [comment]
Looking better now, thanks for the changes.
Still not matching `pt, px`, is it working for you?

Screenshot 2021-12-22 at 17 40 53

**Comment by Fymyte** [comment]
> Looking better now, thanks for the changes.
> Still not matching `pt, px`, is it working for you?
> 
> Screenshot 2021-12-22 at 17 40 53
> 

I have created a new `kittyUnit` syntax group.
By default they are linked to `Type`.
Do prefer to see them as `Number` ?
Modification is [line 40](
https://github.com/fladson/vim-kitty/blob/3e64acc452c84136c610a2c9d881d36fa80a1663/syntax/kitty.vim#L40)

**Comment by fladson** [comment]
Ah, I got it now, let's keep it as it is.
Thank you again for the contribution @Fymyte!

**Pull request merged**